core/types/receipt.go

```diff
@@ -84,11 +84,14 @@ type Receipt struct {
 	BlockNumber      *big.Int    `json:"blockNumber,omitempty"`
 	TransactionIndex uint        `json:"transactionIndex"`
 
-	// OVM legacy: extend receipts with their L1 price (if a rollup tx)
-	L1GasPrice *big.Int   `json:"l1GasPrice,omitempty"`
-	L1GasUsed  *big.Int   `json:"l1GasUsed,omitempty"`
-	L1Fee      *big.Int   `json:"l1Fee,omitempty"`
-	FeeScalar  *big.Float `json:"l1FeeScalar,omitempty"` // always nil after Ecotone hardfork
+	// OVM legacy & Bedrock: extend receipts with their L1 price (if a rollup tx)
+	L1GasPrice          *big.Int   `json:"l1GasPrice,omitempty"`    // L1 Basefee after bedrock
+	L1BlobBaseFee       *big.Int   `json:"l1BlobBaseFee,omitempty"` // always nil prior to the Ecotone hardfork
+	L1GasUsed           *big.Int   `json:"l1GasUsed,omitempty"`
+	L1Fee               *big.Int   `json:"l1Fee,omitempty"`
+	FeeScalar           *big.Float `json:"l1FeeScalar,omitempty"`         // always nil after Ecotone hardfork
+	L1BaseFeeScalar     *big.Float `json:"l1BaseFeeScalar,omitempty"`     // always nil prior to the Ecotone hardfork
+	L1BlobBaseFeeScalar *big.Float `json:"l1BlobBaseFeeScalar,omitempty"` // always nil prior to the Ecotone hardfork
```

Comment:
Why don't we make them `*big.Int`s or even `uint32`s? By the [Ecotone spec](https://github.com/ethereum-optimism/specs/blob/main/specs/protocol/exec-engine.md#ecotone-l1-cost-fee-changes-eip-4844-da) they fit into `uint32`s. This would allow us to avoid dealing with floats and use exact values instead.

Reply:
For the fee scalar this was intentional to keep them as floats, but we can switch to `*f64` from `*big.Float` if we want. I talked with @MSilb7 & pre-ecotone, we already have floats & it is easier to use rather than having to recompute by dividing by 1e6.

Reply:
Yeah the original intent was just to stay consistent with the legacy scalar format, since there was no strong opinion either way.

Since these are net new (so no breaking changes), either format works, as long as it's clear. So if we now have a strong opinion about not being float, that would work.